postreise/plot/plot_powerflow_snapshot.py

```diff
@@ -0,0 +1,342 @@
+    :return: (*pandas.DataFrame*) -- data frame aggregated to the desired precision.
+    """
+    plant_w_xy = project_bus(plant)
+    plant_w_xy["lat"] = plant_w_xy["lat"].round(coordinate_rounding)
+    plant_w_xy["lon"] = plant_w_xy["lon"].round(coordinate_rounding)
+    aggregated = plant_w_xy.groupby(["lat", "lon"]).agg(
+        {"pg": "sum", "x": "mean", "y": "mean"}
+    )
+    return aggregated
+
+
+def plot_powerflow_snapshot(
+    scenario,
+    hour,
+    b2b_dclines=None,
+    demand_centers=None,
+    ac_branch_color="#8B36FF",
+    dc_branch_color="#01D4ED",
+    solar_color="#FFBB45",
+    wind_color="#78D911",
+    demand_color="gray",
+    figsize=(1400, 800),
+    circle_scale_factor=0.25,
+    bg_width_scale_factor=0.001,
+    pf_width_scale_factor=0.00125,
+    arrow_pf_threshold=3000,
+    arrow_dist_threshold=20,
+    num_ac_arrows=1,
+    num_dc_arrows=1,
+    min_arrow_size=5,
+    branch_alpha=0.5,
+    background_map=False,
+    x_range=None,
+    y_range=None,
+    legend_font_size=None,
+):
+    """Plot a snapshot of powerflow.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario to plot.
+    :param str hour: interval to use for snapshot.
+    :param dict b2b_dclines: which DC lines are actually B2B facilities. Keys are:
+        {"from", "to"}, values are iterables of DC line indices to plot (indices in
+        "from" get plotted at the "from" end, and vice versa).
+    :param pandas.DataFrame demand_centers: lat/lon centers at which to plot the demand
+        from each load zone.
+    :param str ac_branch_color: color to plot AC branches.
+    :param str dc_branch_color: color to plot DC branches.
+    :param str solar_color: color to plot solar generation.
+    :param str wind_color: color to plot wind generation.
+    :param str demand_color: color to plot demand.
+    :param tuple(int, int) figsize: size of the bokeh figure (in pixels).
+    :param int/float circle_scale_factor: scale factor for demand/solar/wind circles.
+    :param int/float bg_width_scale_factor: scale factor for grid capacities.
+    :param int/float pf_width_scale_factor: scale factor for power flows.
+    :param int/float arrow_pf_threshold: minimum power flow (MW) for adding arrows.
+    :param int/float arrow_dist_threshold: minimum distance (miles) for adding arrows.
+    :param int num_ac_arrows: number of arrows for each AC branch.
+    :param int num_dc_arrows: number of arrows for each DC branch.
+    :param int/float min_arrow_size: minimum arrow size.
+    :param int/float branch_alpha: opaqueness of branches.
+    :param bool background_map: whether to plot map tiles behind figure.
+    :param tuple(float, float) x_range: x range to zoom plot to (EPSG:3857).
+    :param tuple(float, float) y_range: y range to zoom plot to (EPSG:3857).
+    :param int/str legend_font_size: size to display legend specified as e.g. 12/'12pt'.
+    """
```

Comment:
You can check that the scnario is in analyze state here using `_check_scenario_is_in_analyze_state`